### PR TITLE
add consumption plan as the default option

### PR DESCRIPTION
**Pull request by yiliuTo**
This is pr is to make the AZD template support both ASA consumption plan and standard plan, and the default option will be the consumption plan.

To test it, you can run:
```
mkdir ASA-Samples-Web-Application
cd ASA-Samples-Web-Application
azd auth login
azd config set alpha.springapp on
azd init --template https://github.com/yiliuTo/ASA-Samples-Web-Application/ -b support-con
azd up
```

The template by default uses consumption plan, and if users want to use standard, they can set `azd env set PLAN standard` then run `azd up`. If users have already provisioned the resources with the consumption plan and want to try the Standard plan, they need to run `azd down` first to delete the resources, and then run `azd env set PLAN standard` and `azd up` again to provision and deploy.

### CODE REVIEW

File: infra/main.bicep
Comment (fangjian0423) [inline infra/main.bicep:23]: It can't generate random password in AZD Bicep without KeyVault, and for now, we don't use KeyVault to store psql password (ASA consumption plan don't support MI). Thus we use hard code to set password.

When this template support KeyVault in the future, we should use random password replace of hard code.

File: infra/main.bicep
Comment (fangjian0423) [inline infra/main.bicep:15]: don't let developer choose plan, we can set consumption plan by default. If developer want to use basic/standard plan, they can use `azd env set xxx=true` to switch to basic/standard plan.

**Comment by yiliuTo** [comment]
We will not merge this pr until the corresponding MS doc gets merged. cc @moarychan and @Netyyyy

**Review by fangjian0423 — APPROVED** [review]

**Pull request merged**
